File: 455-assign-cookies/assign-cookies.cpp
class Solution {
public:
    int findContentChildren(vector<int>& g, vector<int>& s) {
        sort(g.begin(), g.end());
        sort(s.begin(), s.end());
        int count = 0;
        int j = 0;
        int i = 0;
        while(j < g.size() && i < s.size()){
            if(s[i] >= g[j]){
                count++;
                j++;
                i++;
            }
            else if(s[i] < g[j]) i++;
        }
        return count;
    }
};

File: 455-assign-cookies/assign-cookies.java
class Solution {
    public int findContentChildren(int[] g, int[] s) {
        int count = 0;
        Arrays.sort(g);
        Arrays.sort(s);
        int i = 0, j = 0;
        while(i < s.length && j < g.length) {
            if(s[i] >= g[j]) {
                count++;
                j++;
                i++;
            }
            else i++;
        }
        return count;
    }
}